curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-localstorage-by-building-a-todo-app/64faca774fd9fd74bc084cc9.md

```diff
@@ -9,14 +9,55 @@ dashedName: step-34
 
 You should display the `Cancel` and `Discard` buttons to the user only if there is some text present in the input fields.
 
-To begin, within the `closeTaskFormBtn` event listener, create a `formInputsContainValues` variable to check if there is a value in the `titleInput` field **or** the `dateInput` field **or** the `descriptionInput` field.
+Start by removing the `confirmCloseDialog.showModal();` line.
+
+Then inside the `closeTaskFormBtn` event listener, write some logic to check if there is a value in the `titleInput`, `dateInput` **or** the `descriptionInput` field.
+
+If any of those input fields has a value, then use the `showModal()` method on `confirmCloseDialog`. Otherwise, if there are no changes, call the `reset()` function to clear the input fields and hide the form modal.
 
 # --hints--
 
-You should use `const` to create a variable `formInputsContainValues` with the value `titleInput.value || dateInput.value || descriptionInput.value;`
+The `confirmCloseDialog` should only show when there's a value in one of the `input` elements. Check your logic for any errors.
```

Comment:
```suggestion
If any of the input fields has a value, then the confirmation dialog should display. Otherwise, the `reset` function should be called.
```